user-management/src/main/scala/com/normation/plugins/usermanagement/api/UserManagementApi.scala

```diff
@@ -300,23 +359,110 @@ class UserManagementApiImpl(
         authzToken: AuthzToken
     ): LiftResponse = {
       (for {
-        user           <- ZioJsonExtractor.parseJson[JsonUserFormData](req).toIO
+        u              <- ZioJsonExtractor.parseJson[JsonUserFormData](req).toIO
+        user            = u.copy(permissions = u.permissions.filter(_ != AuthorizationType.NoRights.id))
+        allRoles       <- RudderRoles.getAllRoles
         checkExistence <- if (!(userService.authConfig.users.keySet contains id)) {
-                            // we may have users that where added by OIDC, and still want to add them in file
+                            // we may have users that where added by other providers, and still want to add them in file if the providers allows extending
                             userRepo.get(id).flatMap {
-                              case Some(u) => userManagementService.add(User(u.id, "", Set()), user.isPreHashed)
-                              case None    => Inconsistency(s"'$id' does not exists").fail
+                              case Some(u) =>
+                                authProvider
+                                  .getProviderProperties()
+                                  .get(u.managedBy)
+                                  .map(_.extendsRoles)
+                                  .map {
+                                    case true  =>
+                                      implicit val currentRoles   = allRoles.values.toSet
+                                      // Unknown permissions are trusted and put in file
+                                      val (roles, authz, unknown) = UserManagementService.parsePermissions(user.permissions.toSet)
+                                      userManagementService
+                                        .add(User(u.id, "", roles.map(_.name) ++ authz.map(_.id) ++ unknown), user.isPreHashed)
+                                    case false =>
+                                      Inconsistency(
+                                        s"Roles of user '$id' cannot be extended by provider '${u.managedBy}' because how the provider was configured"
```

Comment:
This fail if I use `providers = oidc, ldap, file` and try to change the name of an auto provisionned user (ie one without existence in `rudder-user.xml`) but works with an user that exists.  
I think there is two things to change here: 
- 1/ I think the role extension is misunderstood and partially mixed with autoprovisioning. 

An user can always be declared in the `rudder-user.xml`, even if the OIDC provider is configured with autoprovisioning. It's just that it *doesn't* need to, and if it is not, then only properties from the oidc token will be considered. 

Roles can be not at all (`roles.enabled=false`), partially (`roles.enabled=true`, `roles.override=false`), or totally (``roles.enabled=true`, `roles.override=true`) provided by OIDC. But even in the last case, if several providers are configured for Rudder, then you should be able to set role from the UI b/c the idea is that we have a list of fallback, and so, if Bob tried to log-in with OIDC and that failed (server not available for ex), then he can try with file but in that case, he will get the rights that are defined in `rudder-user.xml`for him. These right can be minimal, for ex read-only, because it's a fall back with lesser level of identification. 
This is also used so that there is an administrator that can still log-in even with a main authentication provider down.

But for now, if it's too complicated, we can accept that an auto-provisioned user (ie one that never reached `rudder-user.xml`), we error on role modification (but it would be better to just, in that case, create the user in the xml file, and warn the admin that if oidc is configured for override, what it does will only matter for the fallback case). 

- 2/ I think we need to have two separated form in the user edit: 
    - one for the things in data base (name, email, other fields)
      This one doesn't have anything to do with the authorisation config and is simple crud on base fields
    - one for roles
      The one with these logic checks



Plus, the error is not correctly handled in the elm app, we just get a white page with a red "Error" toast notification